lib/compat/wordpress-6.1/get-global-styles-and-settings.php

```diff
@@ -42,3 +42,77 @@ function ( $item ) {
 		}
 	}
 }
+
+/**
+ * Returns the stylesheet resulting of merging core, theme, and user data.
+ *
+ * @param array $types Types of styles to load. Optional.
+ *                     It accepts 'variables', 'styles', 'presets' as values.
+ *                     If empty, it'll load all for themes with theme.json support
+ *                     and only [ 'variables', 'presets' ] for themes without theme.json support.
+ *
+ * @return string Stylesheet.
+ */
+function gutenberg_get_global_stylesheet( $types = array() ) {
+	// Return cached value if it can be used and exists.
+	// It's cached by theme to make sure that theme switching clears the cache.
+	$can_use_cached = (
+		( empty( $types ) ) &&
+		( ! defined( 'WP_DEBUG' ) || ! WP_DEBUG ) &&
+		( ! defined( 'SCRIPT_DEBUG' ) || ! SCRIPT_DEBUG ) &&
+		( ! defined( 'REST_REQUEST' ) || ! REST_REQUEST ) &&
+		! is_admin()
+	);
+	$transient_name = 'gutenberg_global_styles_' . get_stylesheet();
+	if ( $can_use_cached ) {
+		$cached = get_transient( $transient_name );
+		if ( $cached ) {
+			return $cached;
+		}
+	}
+	$tree                = WP_Theme_JSON_Resolver_Gutenberg::get_merged_data();
+	$supports_theme_json = WP_Theme_JSON_Resolver_Gutenberg::theme_has_support();
+	if ( empty( $types ) && ! $supports_theme_json ) {
+		$types = array( 'variables', 'presets' );
+		if ( current_theme_supports( 'wp-block-styles' ) ) {
+			$types[] = 'block-layout-styles';
+		}
```

Comment:
This file is moved to the 6.1 directory for these set of lines, which switches on outputting layout styles that are used for blocks, if a theme has opted-in to `wp-block-styles` — this way, a theme can still opt-out of the generated layout styles — but we consider them _part_ of the core block styles.